File: control-panel/src/types/student.ts
export type ActivityStatus = "exploring" | "puzzle" | "boss" | "completed" | "offline";
export type BehaviourFlag = "on-track" | "needs-support" | "at-risk";

export interface XpEvent {
  id: string;
  timestamp: string;
  delta: number;
  reason: string;
  updatedBy: string;
}

export interface Student {
  id: string;
  name: string;
  avatarColor: string;
  grade: string;
  lastActive: string;
  currentActivity: string;
  activityStatus: ActivityStatus;
  xp: number;
  xpGoal: number;
  xpEvents: XpEvent[];
  timePlayedMinutes: number;
  streakDays: number;
  behaviour: BehaviourFlag;
  nextCheckpoint: string;
  guardians?: string;
}

File: MathQuest-Capstone/Assets/Scripts/Supabase/Student.cs
using System;
using Supabase;

namespace Supabase {
    [Serializable]
    public class Student
    {
        public string id;          // UUID from Supabase
        public string name;        // Player name
        public string grade;       // Grade
        public string avatar_color;// Avatar color
        public string last_active; // ISO timestamp
        public string current_activity; // Current activity
        public string activity_status;  // Activity status
        public int xp;             // Current XP
        public int xp_goal;        // XP needed for next level
        public int time_played_minutes; // Time played
        public int streak_days;    // Streak days
        public string behaviour;   // "on-track", "at-risk", etc.
        public string next_checkpoint; // Next checkpoint
        public string guardians;   // Guardians info
        public string created_at;  // Created timestamp
        public string updated_at;  // Updated timestamp
        public int level;          // Current level
    }
}